skflow/io/data_feeder.py

```diff
@@ -166,10 +166,22 @@ def __init__(self, X, y, n_classes, batch_size, random_state=None):
         self.input_shape, self.output_shape = _get_in_out_shape(
             self.X.shape, self.y.shape, n_classes, batch_size)
         self.input_dtype, self.output_dtype = self.X.dtype, self.y.dtype
-        if random_state is None:
-            self.random_state = np.random.RandomState(42)
-        else:
-            self.random_state = random_state
+        self.random_state = np.random.RandomState(42) if random_state is None else random_state
+        self.indices = self.random_state.permutation(self.X.shape[0])
+        self.offset = 0
+        self.epoch = 0
+
+    def get_feed_params_fn(self):
```

Comment:
why does this need to be a build function? do you expect this to have parameters?